File: Segundo Curso/1er cuatrimestre/PDOO/Teoria/Pruebas/PeluqueriaRuby/lib/peluqueria.rb
module Basico

  
  class Peluqueria
    
  def initialize
    @citas=Array.new
  end
  
  def dar_cita
    year= 2020+rand(3) #hasta 2 años a partir de 2020
    dia= 1+ rand(31) 
    hora= 9+rand(9)
    cita= Time.new(year, 11, dia, hora)
    @citas<<cita
    cita
  end
  
  def tinte (persona)
    persona.setColorPelo(pelirojo)    
  end
  
end

end

File: Segundo Curso/1er cuatrimestre/PDOO/Teoria/Pruebas/Peluqueria/src/basico/Peluqueria.java

package basico;

import java.time.LocalDateTime;
import java.time.Month;
import java.util.ArrayList;

public class Peluqueria {  //Clase con visibilidad pública. Su constructor es el que ya existe por defecto
    
    ArrayList<LocalDateTime> citas= new ArrayList<>();
       
    LocalDateTime darCita(){ //Da una cita en un día y hora aleatorios
       	int year= (int)(Math.random()*2+2020);	//hasta 2 años a partir del 2020
        int dia= (int) (Math.random()*30+1);
        int hora= (int) (Math.random()*8+9);      
        LocalDateTime cita= LocalDateTime.of(year, Month.OCTOBER, dia, hora, 0);
        citas.add(cita);
        return cita;
    }
    
    public String citaAnulada() {
        return "Cita anulada\n";
    }
    
    void cambiarColor (Persona pers, ColorPelo tinte) {
        pers.SetColorDePelo(tinte);
    }
    
}
